src/Microsoft.DotNet.VersionTools/Dependencies/ProjectJsonUpdater.cs

```diff
@@ -29,33 +29,20 @@ public IEnumerable<BuildInfo> Update(IEnumerable<BuildInfo> buildInfos)
         {
             foreach (string projectJsonFile in ProjectJsonPaths)
             {
-                JObject projectRoot;
+                IEnumerable<BuildInfo> buildInfosUsed = null;
+
                 try
                 {
-                    projectRoot = ReadProject(projectJsonFile);
+                    FileUtils.ReplaceFileContents(projectJsonFile, contents => ReplaceAllDependencyVersions(contents, projectJsonFile, buildInfos, out buildInfosUsed));
                 }
                 catch (Exception e)
                 {
-                    Trace.TraceWarning($"Non-fatal exception occurred reading '{projectJsonFile}'. Skipping file. Exception: {e}. ");
+                    Trace.TraceWarning($"Non-fatal exception occurred reading or writing '{projectJsonFile}'. Skipping file. Exception: {e}. ");
```

Comment:
I don't see anything that could reasonably throw an exception in the code searching for the build info, as it just grabs the Id and Version properties from the LatestPackages PackageIdentity objects, so I think it's fine for the warning message to say that it occurred reading or writing the project.json file.

Agreed about putting the warnings in the PR description as a separate issue.